src/cirrus/builtins/tasks/publish/lambda_function.py

```diff
@@ -1,18 +1,103 @@
 import json
 from os import getenv
 
-from cirrus.lib.logging import get_task_logger
-from cirrus.lib.process_payload import ProcessPayload
-from cirrus.lib.statedb import StateDB
+import boto3
+
+from cirrus.lib2.logging import get_task_logger
+from cirrus.lib2.process_payload import ProcessPayload
+from cirrus.lib2.statedb import StateDB
 
 # envvars
 DATA_BUCKET = getenv("CIRRUS_DATA_BUCKET")
 API_URL = getenv("CIRRUS_API_URL", None)
+PUBLISH_TOPIC_ARN = getenv("CIRRUS_PUBLISH_TOPIC_ARN", None)
 # DEPRECATED - additional topics
 PUBLISH_TOPICS = getenv("CIRRUS_PUBLISH_SNS", None)
 
 # Cirrus state db
 statedb = StateDB()
+snsclient = boto3.client("sns")
+
+
+def sns_attributes(item) -> dict:
+    """Create attributes from Item for publishing to SNS
+
+    Args:
+        item (Dict): A STAC Item
+
+    Returns:
+        Dict: Attributes for SNS publishing
+    """
+    # note that sns -> sqs supports only 10 message attributes
+    # when not using raw mode, and we currently have 10 attrs
+    # possible
+    attr = {
+        "collection": {"DataType": "String", "StringValue": item["collection"]},
+        "bbox.ll_lon": {"DataType": "Number", "StringValue": str(item["bbox"][0])},
+        "bbox.ll_lat": {"DataType": "Number", "StringValue": str(item["bbox"][1])},
+        "bbox.ur_lon": {"DataType": "Number", "StringValue": str(item["bbox"][2])},
+        "bbox.ur_lat": {"DataType": "Number", "StringValue": str(item["bbox"][3])},
+    }
+
+    if "start_datetime" in item["properties"]:
+        attr["start_datetime"] = {
+            "DataType": "String",
+            "StringValue": item["properties"]["start_datetime"],
+        }
+    elif "datetime" in item["properties"]:
+        attr["start_datetime"] = {
+            "DataType": "String",
+            "StringValue": item["properties"]["datetime"],
+        }
+
+    if "end_datetime" in item["properties"]:
+        attr["end_datetime"] = {
+            "DataType": "String",
+            "StringValue": item["properties"]["end_datetime"],
+        }
+    elif "datetime" in item["properties"]:
+        attr["end_datetime"] = {
+            "DataType": "String",
+            "StringValue": item["properties"]["datetime"],
+        }
+
+    if "datetime" in item["properties"]:
+        attr["datetime"] = {
+            "DataType": "String",
+            "StringValue": item["properties"]["datetime"],
+        }
+
+    if "eo:cloud_cover" in item["properties"]:
+        attr["cloud_cover"] = {
```

Comment:
I know this was copied from somewhere else, but I was wondering if it was intentional that the eo:cloud_cover field name is changed to cloud_cover in the event?

Reply:
The name change is because SNS does not support : in attribute names. I’m not sure if dropping the eo is almost appropriate here, but I’d rather not make this a breaking change unless we could preserve the entire attribute name.